lfs/gitscanner_catfilebatch.go

```diff
@@ -0,0 +1,84 @@
+package lfs
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"strconv"
+
+	"github.com/pkg/errors"
+)
+
+// runCatFileBatch uses 'git cat-file --batch' to get the object contents of a
+// git object, given its sha1. The contents will be decoded into a Git LFS
+// pointer. Git Blob SHA1s are read from the sha1Ch channel and fed to STDIN.
+// Results are parsed from STDOUT, and any elegible LFS pointers are sent to
+// pointerCh. Any errors are sent to errCh. An error is returned if the 'git
+// cat-file' command fails to start.
+func runCatFileBatch(pointerCh chan *WrappedPointer, sha1Ch <-chan string, errCh chan error) error {
+	cmd, err := startCommand("git", "cat-file", "--batch")
+	if err != nil {
+		return err
+	}
+
+	go catFileBatchOutput(pointerCh, cmd, errCh)
+	go catFileBatchInput(cmd, sha1Ch, errCh)
+	return nil
+}
+
+func catFileBatchOutput(pointerCh chan *WrappedPointer, cmd *wrappedCmd, errCh chan error) {
+	for {
+		l, err := cmd.Stdout.ReadBytes('\n')
```

Comment:
I think this would make a good follow up PR. Trying not to change things _too much_.